File: library-common/src/main/java/com/drz/common/config/ModuleLifecycleReflexs.java
package com.drz.common.config;

/**
 * 应用模块: common
 * <p>
 * 类描述: 组件生命周期初始化类的类目管理者,在这里注册需要初始化的组件,通过反射动态调用各个组件的初始化方法
 * <p>
 *
 * @author darryrzhoong
 * @since 2020-02-25
 */
public class ModuleLifecycleReflexs
{
    /** 基础库 */
    private static final String BaseInit = "com.drz.common.CommonModuleInit";
    
    /** main组件库 */
    private static final String MainInit =
        "com.drz.main.application.MainModuleInit";

    /**用户组件初始化*/
    private static final String UserInit = "com.drz.user.UserModuleInit";
    
    public static String[] initModuleNames = {BaseInit, MainInit,UserInit};
}

File: library-common/src/main/java/com/drz/common/config/ModuleLifecycleReflexs.kt
package com.drz.common.config

object ModuleLifecycleReflexs {
    private const val BaseInit = "com.drz.common.CommonModuleInit"
    private const val MainInit = "com.drz.main.application.MainModuleInit"
    private const val UserInit = "com.drz.user.UserModuleInit"
    private const val HomeInit = "com.drz.home.HomeModuleInit"
    private const val CommunityInit = "com.drz.community.CommunityModuleInit"
    private const val MoreInit = "com.drz.more.MoreModuleInit"
    private const val PlayerInit = "com.drz.player.PlayerModuleInit"

    val initModuleNames = arrayOf(
        BaseInit, MainInit, UserInit, HomeInit, CommunityInit, MoreInit, PlayerInit
    )
}
